### PR TITLE
[FEATURE] AccessProtectionHelper is multi packages aware !

### PR DESCRIPTION
**Pull request by bastienpaulfr**

### CODE REVIEW

File: core/src/androidTest/java/fr/coppernic/lib/utils/os/AccessProtectionHelperAndroidTest.kt [hunk 1]
```diff
@@ -33,23 +34,39 @@ class AccessProtectionHelperAndroidTest {
...
     }
 
     //748D82E5C019B361C7AD2C6F6AB8587602062EF4F5D444E3A2E44F2D8DCBFAC5
     //E6D539D9E495619B32D32C7CFFB1FEAE1EFA86726CE6F23668E4FB6D1D80F122
     @Test
     fun allowed() {
+        accessProtectionHelper = AccessProtectionHelper(context,
+                mapOf("E6D539D9E495619B32D32C7CFFB1FEAE1EFA86726CE6F23668E4FB6D1D80F122".toLowerCase() to setOf("fr.coppernic.lib.utils.test".toRegex()))
+        )
+
```
Comment (Blacklybarta) [inline core/src/androidTest/java/fr/coppernic/lib/utils/os/AccessProtectionHelperAndroidTest.kt:56]: EL

File: core/src/androidTest/java/fr/coppernic/lib/utils/os/AccessProtectionHelperAndroidTest.kt [hunk 2]
```diff
@@ -33,23 +34,39 @@ class AccessProtectionHelperAndroidTest {
...
         accessProtectionHelper = AccessProtectionHelper(context,
-                HashSet(listOf(
-                        Pair("fr.coppernic.lib.utils.test", "E6D539D9E495619B32D32C7CFFB1FEAE1EFA86726CE6F23668E4FB6D1D80F122".toLowerCase())
-                )))
-        assertThat(accessProtectionHelper.isPackageAllowed(context.packageName), equalTo(true))
+                mapOf(
+                        "E6D539D9E495619B32D32C7CFFB1FEAE1EFA86726CE6F23668E4FB6D1D80F122".toLowerCase()
+                                to setOf(
+                                "or.example.com".toRegex(),
+                                "prout.com".toRegex(),
```
Comment (Blacklybarta) [inline core/src/androidTest/java/fr/coppernic/lib/utils/os/AccessProtectionHelperAndroidTest.kt:67]: attention aux traces

File: core/src/main/java/fr/coppernic/lib/utils/os/AccessProtectionHelper.kt [hunk 3]
```diff
@@ -25,10 +36,10 @@ class AccessProtectionHelper constructor(val context: Context, val whiteList: Ha
      * @return true if process is allowed
      */
     fun isUidAllowed(uid: Int): Boolean {
-        return isPackageAllowed(getPackageForUid(uid))
+        return arePackagesAllowed(getPackageForUid(uid))
     }
 
-    fun getPackageForUid(uid: Int): String {
+    fun getPackageForUid(uid: Int): Set<String> {
```
Comment (Blacklybarta) [inline core/src/main/java/fr/coppernic/lib/utils/os/AccessProtectionHelper.kt]: getPackagesForUid ?

File: core/src/androidTest/java/fr/coppernic/lib/utils/os/AccessProtectionHelperAndroidTest.kt (hunk 2 above)
Comment (bastienpaulfr) [reply core/src/androidTest/java/fr/coppernic/lib/utils/os/AccessProtectionHelperAndroidTest.kt:67]: lol

**Pull request closed without merging**